### PR TITLE
Add the metrics to record the direct memory used by Netty

### PR DESCRIPTION
**Pull request by Jackson-Wang-7**
### What changes are proposed in this pull request?
Add a metric to record the direct memory usage of Netty.

### Why are the changes needed?
when high business pressure, there may be OOM issues when using Netty direct memory to transfer data. Adding this metric to monitor the usage of direct memory.

### Does this PR introduce any user facing changes?
No

### CODE REVIEW

**Comment by yyongycy** [comment]
Overall looks good to me, had you tried somewhere? Any log printed?

File: core/server/master/src/main/java/alluxio/master/throttle/SystemMonitor.java [hunk 1]
```diff
@@ -182,19 +182,19 @@ private void reInitTheThresholds() {
...
         Configuration.getMs(PropertyKey.MASTER_THROTTLE_STRESSED_HEAP_GC_TIME),
-        Configuration.getInt(PropertyKey.MASTER_THROTTLE_STRESSED_RPC_QUEUE_SIZE));
+        Configuration.getInt(PropertyKey.MASTER_THROTTLE_STRESSED_RPC_QUEUE_SIZE), 0);
     // mPitThresholdOverloaded;
     mPitThresholdOverloaded = ServerIndicator.createThresholdIndicator(0,
         Configuration.getDouble(PropertyKey.MASTER_THROTTLE_OVERLOADED_HEAP_USED_RATIO),
         Configuration.getDouble(PropertyKey.MASTER_THROTTLE_OVERLOADED_CPU_LOAD_RATIO),
         Configuration.getMs(PropertyKey.MASTER_THROTTLE_OVERLOADED_HEAP_GC_TIME),
-        Configuration.getInt(PropertyKey.MASTER_THROTTLE_OVERLOADED_RPC_QUEUE_SIZE));
+        Configuration.getInt(PropertyKey.MASTER_THROTTLE_OVERLOADED_RPC_QUEUE_SIZE), 0);
```
[the remark this answers is not in the document]
Comment (Jackson-Wang-7) [reply core/server/master/src/main/java/alluxio/master/throttle/SystemMonitor.java:197]: this is for the threshold of overloaded. currently just set it to 0.

**Review by yyongycy — APPROVED** [review]
lgtm

File: core/common/src/main/java/alluxio/metrics/MetricKey.java [hunk 2]
```diff
@@ -2721,6 +2721,11 @@ public static String getSyncMetricName(long mountId) {
           .setDescription("The used direct memory")
           .setMetricType(MetricType.GAUGE)
           .build();
+  public static final MetricKey PROCESS_NETTY_DIRECT_MEM_USED =
+      new Builder("Process.Netty.direct.mem.used")
```
Comment (jiacheliu3) [inline core/common/src/main/java/alluxio/metrics/MetricKey.java:2725]:
```suggestion
      new Builder("Process.netty.direct.mem.used")
```

File: core/common/src/main/java/alluxio/metrics/MetricKey.java [hunk 3]
```diff
@@ -2721,6 +2721,11 @@ public static String getSyncMetricName(long mountId) {
...
+          .setDescription("The used direct memory of netty")
```
Comment (jiacheliu3) [inline core/common/src/main/java/alluxio/metrics/MetricKey.java:2726]: `Direct memory used by Netty`

File: core/common/src/main/java/alluxio/metrics/MetricKey.java [hunk 4]
```diff
@@ -2721,6 +2721,11 @@ public static String getSyncMetricName(long mountId) {
           .setDescription("The used direct memory")
```
Comment (jiacheliu3) [inline core/common/src/main/java/alluxio/metrics/MetricKey.java:2721]: can we say here `Direct memory used by NIO direct buffer pool`?

**Review by jiacheliu3 — APPROVED** [review]
LGTM with some minor comments, thanks a lot for the work!
Could you add your local printout examples to the PR description? Seeing those numbers really helped me understand :)

**Comment by Jackson-Wang-7** [comment]
```
2024-01-09 11:28:46,626 WARN  [ALLUXIO-PROXY-WEB-SERVICE-265](MetricsSystem.java:187) - mbean direct mem used:180226
2024-01-09 11:28:46,629 WARN  [ALLUXIO-PROXY-WEB-SERVICE-265](MetricsSystem.java:194) - netty direct mem max :104857600
2024-01-09 11:28:46,630 WARN  [ALLUXIO-PROXY-WEB-SERVICE-265](MetricsSystem.java:195) - netty direct mem used:58720542
```
the first line is the number get from mbean of `java.nio:type=BufferPool,name=direct`
the third line is the number about the direct memory used by Netty

**Comment by yyongycy** [comment]
This is to branch-2.10 directly?

**Comment by Jackson-Wang-7** [comment]
> This is to branch-2.10 directly?

@yyongycy @jiacheliu3 I made a mistake in the based branch and created a new PR based on master-2. x #18485

**Pull request closed without merging**